### PR TITLE
ddl: Use testify in ddl/placement_sql_test

### LINKED ISSUE #29107: migrate test-infra to testify for `ddl/placement_sql_test.go`

### PR DESCRIPTION
**Pull request by y5w**
### What problem does this PR solve?

Issue Number: close #29107

Problem Summary:

### What is changed and how it works?
Migrate ddl/placement_sql_test to use testify 

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by y5w** [comment]
I think the PR is ready for review - @tisonkun could you take a look and help me with the pending CI checks. Thanks!

File: ddl/placement_sql_test.go [hunk 1]
```diff
@@ -17,21 +17,28 @@ package ddl_test
...
+	"github.com/stretchr/testify/require"
 )
 
 // TODO: Remove in https://github.com/pingcap/tidb/issues/27971 or change to use SQL PLACEMENT POLICY
-func (s *testDBSuite1) TestPlacementPolicyCache(c *C) {
-	tk := testkit.NewTestKit(c, s.store)
+func TestPlacementPolicyCache(t *testing.T) {
+	store, close := testkit.CreateMockStore(t)
+	defer close()
+	dom, err := session.BootstrapSession(store)
```
Comment (hawkingrei) [inline ddl/placement_sql_test.go:38]: Simple code like this.
```
	store, dom, clean := testkit.CreateMockStoreAndDomain(t)
	defer clean()
```

ditto other.

Comment (y5w) [reply ddl/placement_sql_test.go]: Done

**Review by tisonkun — APPROVED** [review]

**Comment by tisonkun** [comment]
/merge

Despite to https://github.com/pingcap/tidb/pull/32098#discussion_r798727845, the PR itself is clean.

File: ddl/placement_sql_test.go [hunk 2]
```diff
@@ -238,18 +240,22 @@ PARTITION BY RANGE (c) (
...
 	}
 }
 
-func (s *testDBSuite6) TestCreateSchemaWithPlacement(c *C) {
-	tk := testkit.NewTestKit(c, s.store)
+func TestCreateSchemaWithPlacement(t *testing.T) {
+	store, close := testkit.CreateMockStore(t)
+	defer close()
+	dom, err := session.BootstrapSession(store)
+	require.NoError(t, err)
```
Comment (tisonkun) [inline ddl/placement_sql_test.go:257]:
```suggestion
	store, dom, clean := testkit.CreateMockStoreAndDomain(t)
	defer clean()
```

**Comment by y5w** [comment]
@tisonkun @hawkingrei looks like `executor_test.go` is failed with oom error - which seems unrelated to this change. Is it just a flacky test? Could we rerun the unit test maybe?

**Comment by y5w** [comment]
> @tisonkun @hawkingrei looks like `executor_test.go` is failed with oom error - which seems unrelated to this change. Is it just a flacky test? Could we rerun the unit test maybe?

nvm - figured out how to trigger unit test

**Comment by y5w** [comment]
🎉 This is ready to be merged

File: ddl/placement_sql_test.go [hunk 3]
```diff
@@ -17,21 +17,25 @@ package ddl_test
...
 	mysql "github.com/pingcap/tidb/errno"
 	"github.com/pingcap/tidb/parser/model"
 	"github.com/pingcap/tidb/session"
-	"github.com/pingcap/tidb/util/testkit"
+	"github.com/pingcap/tidb/testkit"
 	"github.com/pingcap/tidb/util/testutil"
+	"github.com/stretchr/testify/require"
 )
 
 // TODO: Remove in https://github.com/pingcap/tidb/issues/27971 or change to use SQL PLACEMENT POLICY
```
[the remark this answers is not in the document]
Comment (morgo) [reply ddl/placement_sql_test.go:33]: I think it looks correct in the test and we can remove the comment.

@lcwangchao what do you think?

**Pull request merged**